vnext/Microsoft.ReactNative/Fabric/Composition/TextInput/WindowsTextInputComponentView.cpp

```diff
@@ -811,13 +824,63 @@ void WindowsTextInputComponentView::updateLayoutMetrics(
        layoutMetrics.frame.size.height * layoutMetrics.pointScaleFactor});
 }
 
+void WindowsTextInputComponentView::updateTextColor(COLORREF color) {
+  LRESULT res;
+  CHARFORMAT2 charFormat;
+  ZeroMemory(&charFormat, sizeof(CHARFORMAT2));
+  charFormat.cbSize = sizeof(CHARFORMAT2);
+  charFormat.dwMask = CFM_COLOR;
+  charFormat.crTextColor = color;
+
+  // Set the color of the text
+  winrt::check_hresult(m_textServices->TxSendMessage(EM_SETCHARFORMAT, 0, reinterpret_cast<LPARAM>(&charFormat), &res));
+}
+
+std::string WindowsTextInputComponentView::findExtraChar(
+    std::string const &originalString,
+    std::string const &modifiedString) {
+  auto newLength = modifiedString.size() - originalString.size();
+  size_t i = 0;
+  // find first location of extra char
+  while (i < modifiedString.size()) {
+    if (originalString[i] != modifiedString[i]) {
+      break;
+    }
+    i++;
+  }
+  return modifiedString.substr(i, newLength);
+}
+
 // When we are notified by RichEdit that the text changed, we need to notify JS
 void WindowsTextInputComponentView::OnTextUpdated() noexcept {
   auto data = m_state->getData();
   // auto newAttributedString = getAttributedString();
   // if (data.attributedString == newAttributedString)
   //    return;
   data.attributedString = getAttributedString();
+
+  // if the string is just m_placholderText, don't notify the JS
+  if (data.attributedString.getFragments().size() &&
+      data.attributedString.getFragments()[0].string == m_placeholderText) {
```

Comment:
Dont we need to notify the JS when the text becomes "empty", even if we are displaying the placeholder text?